webpack/assets/javascripts/foreman_tools.js

```diff
@@ -10,6 +10,36 @@ import { sprintf, translate as __ } from './react_app/common/I18n';
 
 import { showLoading, hideLoading } from './foreman_navigation';
 
+import store from './react_app/redux';
+import { openConfirmModal as coreOpenConfirmModal } from './react_app/components/ConfirmModal';
+
+export const openConfirmModal = options =>
+  store.dispatch(coreOpenConfirmModal(options));
+
+if ($.rails) {
+  // override the jQuery UJS $.rails.allowAction
+  $.rails.allowAction = element => {
+    const message = element.data('confirm');
+    const isWarning = element.data('method') === 'delete';
+    if (!message) return true;
+
+    if ($.rails.fire(element, 'confirm')) {
+      openConfirmModal({
+        title: __('Confirm'),
+        message,
+        isWarning,
+        onConfirm: () => {
+          const oldAllowAction = $.rails.allowAction;
+          $.rails.allowAction = () => true;
+          element.trigger('click');
+          $.rails.allowAction = oldAllowAction;
+        },
+      });
+    }
+    return false;
+  };
+}
```

Comment:
Will this run every time that `foreman_tools` is imported? Perhaps we should move it to a separate `rails_overrides.js` file that is imported once in the bundle?

Reply:
Moved it to application.js fie's `onContentLoad`
although it was actually called only once each full-refresh in foreman_tools, same as in application.js
it feels that it's a better place to store such overrides :)

Reply:
moved to `foreman_overrides.js` instead